apps/system/js/app_usage_metrics.js

```diff
@@ -318,7 +334,23 @@
         }
 
         // Move on to the next step in the startup process
-        getConfigurationSettings();
+        waitForApplicationsReady();
+      });
+    }
+
+    function getImeiCode() {
+      var dialPromise = navigator.mozTelephony.dial('*#06#', 0);
+      return dialPromise.then(function about_dialImeiPromise(call) {
+        return call.result.then(function (result) {
+          if (result && result.success && (result.serviceCode === 'scImei')) {
+            asyncStorage.setItem(IMEI_KEY, result.statusMessage);
+            return result.statusMessage;
+          } else {
+            var errorMsg = 'Could not retrieve the IMEI code for SIM ';
+            debug(errorMsg);
+            return Promise.reject(new Error(errorMsg));
```

Comment:
Have you tested the case when this promise is rejected? It looks like the rejection won't get bubbled up, meaning that getDeviceID() will never be called up on line 311